### DISCUSSION TITLE
How to create new directory in repository via Bash in starter workflow?

### DISCUSSION THREAD

**Discussion by EmilyGraceSeville7cf** [community/community#26048] [Actions]
<p>Hello! Could somebody help me to create new directory in repository via Bash?</p>
<pre><code class="lang-auto"># This is a basic workflow to help you get started with Actions

name: CI

# Controls when the action will run. Triggers the workflow on push or pull request
# events but only for the $default-branch branch
on:
  push:
    branches: [ $default-branch ]
  pull_request:
    branches: [ $default-branch ]

# A workflow run is made up of one or more jobs that can run sequentially or in parallel
jobs:
  # This workflow contains a single job called "build"
  build:
    # The type of runner that the job will run on
    runs-on: ubuntu-latest

    # Steps represent a sequence of tasks that will be executed as part of the job
    steps:
      # Checks-out your repository under $GITHUB_WORKSPACE, so your job can access it
      - uses: actions/checkout@v2

      # Runs a single command using the runners shell
      - name: Run a one-line script
        run: # What to write here? Simple "mkdir &lt;name&gt;" doesn't work.
</code></pre>

**Comment by BrightRan** [comment]
<p><a class="mention" href="/alvin-seville">@alvin-seville</a>,</p>
<p>You just need to use the <strong>mkdir</strong> command to create the new directory via Bash shell.<br>
Basic syntax:</p>
<pre><code class="lang-yaml">mkdir [options] &lt;name_of_directory&gt;
</code></pre>
<ol>
<li>
<p>The <strong>&lt;name_of_directory&gt;</strong> can be a relative path to the current working directory.</p>
<ul>
<li>
<p>Create a single level sub-directory in current working directory:</p>
<pre><code class="lang-bash">mkdir subdir1
</code></pre>
</li>
<li>
<p>Create multiple levels sub-directory in current working directory, the ‘<strong>-p</strong>’ option is required:</p>
<pre><code class="lang-bash">mkdir -p subdir1/subdir2
</code></pre>
</li>
</ul>
</li>
<li>
<p>The <strong>&lt;name_of_directory&gt;</strong> also can be an absolute path of the new directory, the ‘<strong>-p</strong>’ option also is required when creating multiple levels directory.</p>
<pre><code class="lang-bash">mkdir -p /home/subdir1/subdir2
</code></pre>
</li>
</ol>

**Comment by Simran-B** [comment, marked as answer]
<p>What do you mean by “a simple mkdir doesn’t work”?<br>
It should create the directory, but if you want to commit that to the repository, then you must place a file inside. Git doesn’t support tracking empty folders.</p>
<p>A common convention is to place a file called <code>.gitkeep</code> in such placeholder folders, either without content or with a brief description of what this file is for.</p>
